subsys/bluetooth/controller/hci_driver.c

```diff
@@ -32,6 +32,20 @@ static K_THREAD_STACK_DEFINE(recv_thread_stack, CONFIG_BLECTLR_RX_STACK_SIZE);
 static K_THREAD_STACK_DEFINE(signal_thread_stack,
 			     CONFIG_BLECTLR_SIGNAL_STACK_SIZE);
 
+
+/* It should not be possible to set CONFIG_BLECTRL_SLAVE_COUNT larger than
+ * CONFIG_BT_MAX_CONN. Kconfig should make sure of that, this assert is to
+ * verify that assumption. */
+BUILD_ASSERT(CONFIG_BLECTRL_SLAVE_COUNT <= CONFIG_BT_MAX_CONN);
+
+#define BLECTRL_MASTER_COUNT (CONFIG_BT_MAX_CONN - CONFIG_BLECTRL_SLAVE_COUNT)
+
+BUILD_ASSERT((IS_ENABLED(CONFIG_BT_CENTRAL) && (CONFIG_BLECTRL_SLAVE_COUNT > 0)) ||
+			 !IS_ENABLED(CONFIG_BT_CENTRAL));
+
+BUILD_ASSERT((IS_ENABLED(CONFIG_BT_PERIPHERAL) && (BLECTRL_MASTER_COUNT > 0)) ||
+			 !IS_ENABLED(CONFIG_BT_PERIPHERAL));
```

Comment:
!IS_ENABLED part does not change the result.

Reply:
It does, see reasoning in other comment.

Reply:
Yes. But still can be simplified like in other comment.